app/Libraries/Api/Models/BaseApiModel.php

```diff
@@ -113,27 +118,70 @@ public function __construct(array $attributes = [])
 
     public function scopePublished(Builder $query): Builder
     {
-        //
+        if (config('aic.is_preview_mode')) {
+            return $query;
+        }
+
+        // scopeVisible checks for its fillable columns
+        $query->visible();
+
+        if ($this->isFillable('published')) {
+            $query->wherePublished(true);
+        }
+
+        return $query;
     }
 
     public function scopeAccessible(Builder $query): Builder
     {
-        //
+        if (! TwillPermissions::enabled()) {
+            return $query;
+        }
+
+        $model = get_class($query->getModel());
+        $moduleName = TwillPermissions::getPermissionModule(getModuleNameByModel($model));
+
+        if ($moduleName && ! Auth::user()->isSuperAdmin()) {
+            // Get all permissions the logged in user has regards to the model.
+            $allPermissions = Auth::user()->allPermissions();
+            $allModelPermissions = (clone $allPermissions)->ofModel($model);
+
+            // If the user has any module permissions, or global manage all modules permissions, all items will be return
+            if (
+                (clone $allModelPermissions)->module()
+                    ->whereIn('name', Permission::available(Permission::SCOPE_MODULE))
+                    ->exists()
+                || (clone $allPermissions)->global()->where('name', 'manage-modules')->exists()
+            ) {
+                return $query;
+            }
+
+            // If the module is submodule, skip the scope.
+            if (strpos($moduleName, '.')) {
+                return $query;
+            }
+
+            $authorizedItemsIds = $allModelPermissions->moduleItem()->pluck('permissionable_id');
+
+            return $query->whereIn($this->getTable() . '.id', $authorizedItemsIds);
+        }
+
+        return $query;
     }
 
     public function scopeOnlyTrashed(Builder $query): Builder
     {
-        //
+        return $query->where("{$this->getTable()}.published", false);
     }
 
     public function scopeDraft(Builder $query): Builder
     {
-        //
+        return $query->onlyTrashed();
     }
```

Comment:
The bodies of these methods should be swapped, yea?

Reply:
Yup! Nice catch.